r/tools/nixlibs.R

```diff
@@ -22,7 +22,14 @@ dst_dir <- paste0("libarrow/arrow-", VERSION)
 # TESTING is set in test-nixlibs.R; it won't be set when called from configure
 test_mode <- exists("TESTING")
 
-arrow_repo <- paste0(getOption("arrow.dev_repo", "https://nightlies.apache.org/arrow/r"), "/libarrow/")
+# Check if version string has 4th component
```

Comment:
There's one other case that we should support here. Suppose we submit 9.0.0 to CRAN and they reject it (as happens about 50% of the time). Assume the fix is something in the R package only. We tweak that and resubmit as 9.0.0.1. We want the official artifactory 9.0.0 libarrow to be used. 

Here's some old code that did some inspection of the 4th component and did different things: https://github.com/apache/arrow/blob/6c7c4f0a09a3f40d65999511c77d0f21473b4de8/r/tools/nixlibs.R#L216-L233 We don't want exactly that, but maybe something that downloads from artifactory if `is.na(p) || p < 100` or something like that?

Reply:
Ah, should we harmonize the behavior between nix and windows as winlibs already has this? I think it makes sense to use you proposed idea for both as going forward all releases will have binaries uploaded to the artifactory. (So no cutting off patch versions etc.)